tailscale/rootfs/etc/s6-overlay/s6-rc.d/proxy/run

```diff
@@ -25,10 +26,18 @@ fi
 
 # Set up proxy
 # Note: Configuration is persistent between reboots, though configuring the same value won't cause duplicate entries,
-#       but will overwite previous entry if core.port has been changed
+#       but will overwrite the previous entry if core.port has been changed
 # Note: If a tailnet node is renamed, there will be deprecated entries in status json under .Web.<different-domain:443>,
 #       those can't be removed, no command for that situation currently
-if ! /opt/tailscale serve https:443 / "http://127.0.0.1:$(bashio::core.port)"; then
+
+# Check if proxy_target_port is configured in the add-on settings
+if bashio::config.has_value 'proxy_target_port'; then
+  proxy_target_port=$(bashio::config 'proxy_target_port')
+else
+  proxy_target_port=$(bashio::core.port)
+fi
+
+if ! /opt/tailscale serve https:443 / "http://127.0.0.1:${proxy_target_port}"; then
```

Comment:
If you redirect the target of the tailscale serve command to another httpS port, it won't work (tailscale serve can accept self signed certs, but can't accept "invalid" certs, like DuckDNS cert behind 127.0.0.1:target_port). So tailscale serve will work only, when the NGINX reverse proxy is a plain HTTP proxy.

And the current version of the PR checks HA to be on a HTTP port, but then redirects tailscale serve to another "unchecked" port, it seems to me inconsistent.

Reply:
Good points. One option would be to change the serve command to include `+insecure` which I believe would overcome "invalid" certs. All that said I think this needs a lot of consideration.

Reply:
No, `+insecure` won't work. Tested it 1 year ago, and now again, same problem, different error message.

Insecure seems to be for self signed certs only, `https+insecure://127.0.0.1:xxx` or `https://127.0.0.1:xxx` works the same in case of a real cert behind 127.0.0.1:xxx:
- browser receives 502
- tailscale logs "http: proxy error: remote error: tls: unrecognized name" a year ago the message was more useful: "proxy error: x509: cannot validate certificate for 127.0.0.1 because it doesn't contain any IP SANs"
- because it will not accept a valid (eg. Let's Encrypt) cert for a server on 127.0.0.1, and you can't change 127.0.0.1 in tailscale serve, and you can't add localhost to certs (if I know correctly)

I don't know whether it is a tailscale bug, or I did something wrong. If you are successful to target with tailscale proxy a https server, please give some example.

Reply:
I don't have easy access to a test environment for this setup at the moment, but perhaps the [TCP forwarder](https://tailscale.com/kb/1242/tailscale-serve/#tcp-forwarder) would work? Ex: `serve tcp:443 tcp://127.0.0.1:xxx`

Reply:
tcp:443 will result in ERR_SSL_UNRECOGNIZED_NAME_ALERT